src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlStatisticsAggregateMethodTranslator.cs

```diff
@@ -39,18 +39,66 @@ public class NpgsqlStatisticsAggregateMethodTranslator : IAggregateMethodCallTra
             _ => null
         };
 
-        if (functionName is null)
+        if (functionName is not null)
         {
-            return null;
+            return source.Selector is SqlExpression sqlExpression
+                ? _sqlExpressionFactory.AggregateFunction(
+                    functionName,
+                    new[] { sqlExpression },
+                    source,
+                    nullable: true,
+                    argumentsPropagateNullability: new[] { false },
+                    typeof(double),
+                    _doubleTypeMapping)
+                : null;
+        }
+
+        functionName = method.Name switch
+        {
+            nameof(NpgsqlAggregateDbFunctionsExtensions.Correlation) => "corr",
+            nameof(NpgsqlAggregateDbFunctionsExtensions.CovariancePopulation) => "covar_pop",
+            nameof(NpgsqlAggregateDbFunctionsExtensions.CovarianceSample) => "covar_samp",
+            nameof(NpgsqlAggregateDbFunctionsExtensions.RegrAverageX) => "regr_avgx",
+            nameof(NpgsqlAggregateDbFunctionsExtensions.RegrAverageY) => "regr_avgy",
+            nameof(NpgsqlAggregateDbFunctionsExtensions.RegrCount) => "regr_count",
+            nameof(NpgsqlAggregateDbFunctionsExtensions.RegrIntercept) => "regr_intercept",
+            nameof(NpgsqlAggregateDbFunctionsExtensions.RegrR2) => "regr_r2",
+            nameof(NpgsqlAggregateDbFunctionsExtensions.RegrSlope) => "regr_slope",
+            nameof(NpgsqlAggregateDbFunctionsExtensions.RegrSXX) => "regr_sxx",
+            nameof(NpgsqlAggregateDbFunctionsExtensions.RegrSXY) => "regr_sxy",
+            _ => null
+        };
+
+        if (functionName is not null)
+        {
+            // These methods accept two enumerable (column) arguments; this is represented in LINQ as a projection from the grouping
+            // to a tuple of the two columns. Since we generally translate tuples to PostgresRowValueExpression, we take it apart here.
+            if (source.Selector is not PostgresRowValueExpression rowValueExpression)
```

Comment:
@smitpatel here's a translation of an aggregate function with multiple enumerable arguments. See test below for the C# side - as discussed the grouping needs to be projected to a tuple containing the two columns. It looks a bit weird/complex, but it prevents applying different predicate/orderings/distincts which would have been possible had the function accepted two enumerables instead.

What do you think, how does it look?

Funnily enough, since the PG provider translates ValueTuple to PostgresRowValueExpression, that's what we get here. That's also a bit weird, but seems OK.

Reply:
I do have some idea, let's discuss next time we are in office.

Reply:
BTW @smitpatel at some point we can discuss this too